initramfs/src/registry.rs

```diff
@@ -1,97 +1,311 @@
+            }
+            ManifestType::OCIv1List => {
+                let manifest = res
+                    .json::<OCIv1ListResponse>()
+                    .await
+                    .map_err(|e| anyhow!(e).context("Failed to parse response"))?;
+
+                self.manifest_from_oci_list(manifest, image_name).await?
+            }
+            ManifestType::OCIv1 => unimplemented!(),
+        };
+
+        let layers_metadata = manifest
+            .layers
+            .into_iter()
+            .map(|layer| LayerMetadata {
+                blob_sum: layer.digest.clone(),
+            })
+            .collect::<Vec<_>>();
+
+        debug!(
+            "Successfully got layers : \n\t{}",
+            layers_metadata
+                .iter()
+                .map(|layer| layer.blob_sum.clone())
+                .collect::<Vec<String>>()
+                .join("\n\t")
+        );
 
         Ok(layers_metadata)
     }
 
-    async fn get_layer(
+    async fn manifest_from_manifest_v2_list(
+        &self,
+        manifest: ManifestListV2Response,
+        image_name: &str,
+    ) -> Result<ManifestV2Response> {
+        let sub_manifest = manifest
+            .manifests
+            .iter()
+            .find(|manifest| {
+                manifest
+                    .platform
+                    .as_ref()
+                    .map(|platform| {
+                        platform.os == DEFAULT_PLATFORM_OS
+                            && platform.architecture == DEFAULT_PLATFORM_ARCHITECTURE
+                    })
+                    .unwrap_or(false)
+            })
+            .map(|manifest| manifest.digest.clone())
+            .ok_or(anyhow!("No manifest found"))?;
+
+        trace!("Sub manifest : {:?}", sub_manifest);
+
+        self.manifest_from_oci(sub_manifest, image_name).await
+    }
+
+    async fn manifest_from_oci_list(
+        &self,
+        oci_manifest: OCIv1ListResponse,
+        image_name: &str,
+    ) -> Result<ManifestV2Response> {
+        let sub_manifest = oci_manifest
+            .manifests
+            .into_iter()
+            .find(|manifest| {
+                manifest
+                    .platform
+                    .as_ref()
+                    .map(|platform| {
+                        platform.os == DEFAULT_PLATFORM_OS
+                            && platform.architecture == DEFAULT_PLATFORM_ARCHITECTURE
+                    })
+                    .unwrap_or(false)
+            })
+            .ok_or(anyhow!("No manifest found"))?;
+
+        trace!("Sub manifest : {:?}", sub_manifest);
+
+        self.manifest_from_oci(sub_manifest.digest, image_name)
+            .await
+    }
+
+    async fn manifest_from_oci(
         &self,
-        token: &str,
+        oci_manifest: String,
         image_name: &str,
-        layer_metadata: &LayerMetadata,
-    ) -> Result<Layer> {
+    ) -> Result<ManifestV2Response> {
+        debug!("Downloading manifest for linux/amd64");
```

Comment:
Could you just change that to use the consts `DEFAULT_PLATFORM_OS` and `DEFAULT_PLATFORM_ARCHITECTURE` instead?

Reply:
Done !